### DISCUSSION TITLE
Cannot autowire service ... you should configure its value explicitly

### DISCUSSION THREAD

**Discussion by CJDennis** [symfony/symfony#46150] [Q&A] [answered]
I'm trying to add an autowire parameter to one of my services but it's giving me an error despite the same parameter working fully in a different service.
```
Cannot autowire service "hmr.service.payment": argument "$eway" of method "HMRX\CoreBundle\Services\PaymentService::__construct()" is type-hinted "array", you should configure its value explicitly.
```
If I remove `array` from the parameter, I get a different error message instead.
```
Cannot autowire service "hmr.service.payment": argument "$eway" of method "HMRX\CoreBundle\Services\PaymentService::__construct()" has no type-hint, you should configure its value explicitly.
```
What am I missing?

**Comment by ghost** [comment]
`array` type hint doesn't mean anything. You have to explicitly configure what Symfony DI should pass. There are few ways to do that depending on the SF version you are using. You probably want tagged iterator
https://symfony.com/doc/current/service_container/tags.html

**Comment by CJDennis** [comment, marked as answer]
I found a second definition for my service (in a different file) that was clobbering my main one. Removing the second one caused the changes to start working!
